compiler/z/codegen/S390GenerateInstructions.cpp

```diff
@@ -2051,24 +2051,38 @@ TR::Instruction * generateVRRiInstruction(
 /* Note subtle differences between register types and optionality of masks between these 3*/
 TR::Instruction *
 generateVRSaInstruction(TR::CodeGenerator * cg, TR::InstOpCode::Mnemonic op, TR::Node * n, TR::Register * targetReg, TR::Register * sourceReg, TR::MemoryReference * mr,
-                        uint8_t mask4 /* 4 bits */)
+                        uint8_t mask4 /* 4 bits */, TR::Instruction* preced)
    {
-   return new (INSN_HEAP) TR::S390VRSaInstruction(cg, op, n, targetReg, sourceReg, mr, mask4);
+   if (mr != NULL) preced = mr->enforceSSFormatLimits(n, cg, preced);
+
+   if (preced)
```

Comment:
A comment on the use of two different constructors here because I can't find a better place to put it. We have to use two constructors here for a lot of instruction formats because the root `OMR::Instruction` has two constructors and they use different `preceding instructions`. We can misplace an instruction if the `preceding instruction` is not used properly.